File: Week4/HW2/Pages/basketpage.py
import time


class BasketPage():
    def __init__(self, driver):
        self.driver = driver

        self.scroll_to_top = self.driver.execute_script("scroll(0, 0);")
        self.basket_button = self.driver.find_element_by_xpath("//span[@class='small--hide']")



    def goto_basket(self):
        time.sleep(2)

        self.basket_button.click()

    def remove_from_basket(self):
        self.driver.find_element_by_css_selector("tbody > tr:nth-of-type(1) .cart__remove").click()

    def check_item_number_at_basket(self):
        return int(self.driver.find_element_by_xpath('//*[@id="CartCount"]').get_attribute("textContent"))

    def item_quantity_in_basket_element(self):
        return self.driver.find_element_by_xpath("//input[@name='updates[]']")

    def per_item_price(self):
        return self.driver.find_element_by_xpath("//*[@id='MainContent']/div[1]/div/form/table/tbody/tr/td[3]").get_attribute("textContent").replace("\n","").replace("TL","")

    def refresh_basket(self):
        self.refresh_button = self.driver.find_element_by_xpath("//button[@name='update']")
        self.refresh_button.click()

File: Week4/HW1/PageObject/BasketPage.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Remote;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Interactions;
using NUnit.Framework;
using dotNET_Test.TestClasses;
using SeleniumExtras.PageObjects;

namespace dotNET_Test.PageObjects {

public class BasketPage :   BasePage {

    public BasketPage (IWebDriver driver) {
        BasketPage.driver = driver;
        PageFactory.InitElements(driver,this);
      }
    public void remove_from_basket()
    {
        driver.FindElement(By.CssSelector("tbody > tr:nth-of-type(1) .cart__remove")).Click();
    }
    
    public String check_item_number_at_basket()
    {
        return driver.FindElement(By.XPath("//*[@id='CartCount']")).GetAttribute("textContent");
    }

    public void change_item_quantity(String newQuantity)
    {
        IWebElement quantity = driver.FindElement(By.XPath("//input[@name='updates[]']"));
        quantity.Clear();
        quantity.SendKeys(newQuantity);
        refresh_basket();

    }

    public void refresh_basket()
    {
        driver.FindElement(By.XPath("//button[@name='update']")).Click();
    }

}
}   